src/plugins/memdump/memdump.cpp

```diff
@@ -0,0 +1,252 @@
+    fp = fopen(file, "w");
+
+    if (!fp)
+    {
+        PRINT_DEBUG("[MEMDUMP] Failed to open file\n");
+        goto done;
+    }
+
+    for (size_t i = 0; i < num_pages; i++)
+    {
+        if (access_ptrs[i])
+        {
+            fwrite(access_ptrs[i], VMI_PS_4KB, 1, fp);
+            munmap(access_ptrs[i], VMI_PS_4KB);
+        }
+        else
+        {
+            // unaccessible page, pad with zeros to ensure proper alignment of the data
+            uint8_t zeros[VMI_PS_4KB];
+            fwrite(zeros, VMI_PS_4KB, 1, fp);
+        }
+    }
+
+    fclose(fp);
+    ret = true;
+
+done:
+    if (file)
+        free(file);
+
+    if (access_ptrs)
+        g_free(access_ptrs);
+
+    return ret;
+}
+
+static event_response_t free_virtual_memory_hook_cb(drakvuf_t drakvuf, drakvuf_trap_info_t* info)
+{
+    // HANDLE ProcessHandle
+    uint64_t process_handle = drakvuf_get_function_argument(drakvuf, info, 1);
+    // OUT PVOID *BaseAddress
+    addr_t mem_base_address_ptr = drakvuf_get_function_argument(drakvuf, info, 2);
+
+    if (process_handle != 0xffffffffffffffffULL)
+    {
+        PRINT_DEBUG("[MEMDUMP] Process handle not pointing to self, ignore\n");
+        return VMI_EVENT_RESPONSE_NONE;
+    }
+
+    auto plugin = get_trap_plugin<memdump>(info);
+    if (!plugin)
+        return VMI_EVENT_RESPONSE_NONE;
+
+    vmi_instance_t vmi = drakvuf_lock_and_get_vmi(drakvuf);
+    access_context_t ctx = { .translate_mechanism = VMI_TM_PROCESS_DTB, .dtb = info->regs->cr3 };
+    ctx.addr = mem_base_address_ptr;
+
+    addr_t mem_base_address;
+
+    if (VMI_SUCCESS != vmi_read_addr(vmi, &ctx, &mem_base_address))
+    {
+        PRINT_DEBUG("[MEMDUMP] Failed to read base address in NtFreeVirtualMemory\n");
+        drakvuf_release_vmi(drakvuf);
+        return VMI_EVENT_RESPONSE_NONE;
+    }
+
+    mmvad_info_t mmvad;
+
+    if (VMI_SUCCESS != drakvuf_find_mmvad(drakvuf, info->proc_data.base_addr, mem_base_address, &mmvad))
+    {
+        PRINT_DEBUG("[MEMDUMP] Failed to find MMVAD for memory passed to NtFreeVirtualMemory\n");
+        drakvuf_release_vmi(drakvuf);
+        return VMI_EVENT_RESPONSE_NONE;
+    }
+
+    ctx.addr = mem_base_address;
+    uint16_t magic;
+    char* magic_c = (char*)&magic;
+
+    if (VMI_SUCCESS != vmi_read_16(vmi, &ctx, &magic))
+    {
+        PRINT_DEBUG("[MEMDUMP] Failed to access memory to be used with NtFreeVirtualMemory\n");
+        drakvuf_release_vmi(drakvuf);
+        return VMI_EVENT_RESPONSE_NONE;
+    }
+
+    if (magic_c[0] == 'M' && magic_c[1] == 'Z')
+    {
+        ctx.addr = mmvad.starting_vpn << 12;
+        size_t num_pages = mmvad.ending_vpn - mmvad.starting_vpn + 1;
+        printf("[MEMDUMP] Possible binary detected at address %llx in PID %d\n", (unsigned long long)ctx.addr, info->proc_data.pid);
+        if (!plugin->memdump_dir)
+        {
+            printf("[MEMDUMP] Not saving memory dump because --memdump-dir was not provided\n");
```

Comment:
This should be a `PRINT_DEBUG`

Reply:
Or actually may be a useful printout (provided it supports all output options) that a potential target for the memdump was detected but not saved